server/lib/routes/auth.js

```diff
@@ -0,0 +1,88 @@
+const router = require('express').Router();
+const client = require('../db-client.js');
+const bcrypt = require('bcryptjs');
+const jwt = require('../jwt');
+
+function getProfileWithToken(profile) {
+  return {
+    id: profile.id,
+    username: profile.username,
+    token: jwt.sign({ id: profile.id })
+  };
+}
+
+router
+  .post('/signup', (req, res) => {
+    const body = req.body;
+    const username = body.username;
+    const first = body.firstname;
+    const last = body.lastname;
+    const email = body.email;
+    const password = body.password;
+
+    if(!username || !password) {
+      res.status(400).json({ error: 'username and password required' });
+      return;
+    }
+
+    client.query(`
+      SELECT id
+      FROM profile
+      WHERE username = $1;
+    `,
+    [username])
+      .then(result => {
+        if(result.rows.length > 0) {
+          res.status(400).json({ error: 'username already exists' });
+          return;
+        }
+
+        console.log('creating new user profile...');
+
+        //insert the new user into profile
+        client.query(`
+          INSERT into profile (username, first, last, email, password)
```

Comment:
```suggestion
          INSERT into profile (username, first, last, email, hash)
```
Remember the profile table was updated to include a hash column ^.